exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/traces/SpanStatusStatelessMarshaler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.otlp.traces;
+
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.exporter.internal.marshal.MarshalerContext;
+import io.opentelemetry.exporter.internal.marshal.MarshalerUtil;
+import io.opentelemetry.exporter.internal.marshal.ProtoEnumInfo;
+import io.opentelemetry.exporter.internal.marshal.Serializer;
+import io.opentelemetry.exporter.internal.marshal.StatelessMarshaler;
+import io.opentelemetry.proto.trace.v1.internal.Status;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.io.IOException;
+
+final class SpanStatusStatelessMarshaler implements StatelessMarshaler<StatusData> {
+  static final SpanStatusStatelessMarshaler INSTANCE = new SpanStatusStatelessMarshaler();
+
+  @Override
+  public void writeTo(Serializer output, StatusData status, MarshalerContext context)
+      throws IOException {
+    ProtoEnumInfo protoStatusCode = Status.StatusCode.STATUS_CODE_UNSET;
+    if (status.getStatusCode() == StatusCode.OK) {
+      protoStatusCode = Status.StatusCode.STATUS_CODE_OK;
+    } else if (status.getStatusCode() == StatusCode.ERROR) {
+      protoStatusCode = Status.StatusCode.STATUS_CODE_ERROR;
+    }
+
+    byte[] descriptionUtf8 = context.getData(byte[].class);
+
+    output.serializeString(Status.MESSAGE, descriptionUtf8);
+    output.serializeEnum(Status.CODE, protoStatusCode);
+  }
+
+  @Override
+  public int getBinarySerializedSize(StatusData status, MarshalerContext context) {
+    ProtoEnumInfo protoStatusCode = Status.StatusCode.STATUS_CODE_UNSET;
+    if (status.getStatusCode() == StatusCode.OK) {
+      protoStatusCode = Status.StatusCode.STATUS_CODE_OK;
+    } else if (status.getStatusCode() == StatusCode.ERROR) {
+      protoStatusCode = Status.StatusCode.STATUS_CODE_ERROR;
+    }
```

Comment:
Let's breakout a helper method for computing the `ProtoEnumInfo protoStatusCode`:
```
  private static ProtoEnumInfo toProtoStatusCode(StatusData status) {
    if (status.getStatusCode() == StatusCode.OK) {
      return Status.StatusCode.STATUS_CODE_OK;
    } else if (status.getStatusCode() == StatusCode.ERROR) {
      return Status.StatusCode.STATUS_CODE_ERROR;
    }
    return Status.StatusCode.STATUS_CODE_UNSET;
  }
```